### PR TITLE
Added redirect for students and TAs

### PR DESCRIPTION
**Pull request by WeissSa**
Added a redirect so that students and TAs do not get a 403 error upon going to <course>/courses/:id

## Motivation and Context
This solves a 403 error upon accessing courses through Quercus (I think)

## Your Changes
I added a redirect to the root_url in courses_controller. This will redirect unauthorized users to the root_url. Which for students and TA's is the assignments page and for unauthorized users should be the login page.

Also fixed tests
**Description**:

**Type of change** (select all that apply):

- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Test update (change that modifies or updates tests only)  

## Testing
I tried it as a student, instructor, and TA as well as as a user who is not logged in. It also fails 2 tests (which is good, since those tests expect a 403 error).

## Questions and Comments (if applicable)
Should I remove the tests or modify the tests which expect 403 error.

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have verified that the pre-commit.ci checks have passed. 
- [x] I have verified that the CI tests have passed. 
- [x] I have reviewed the test coverage changes reported on Coveralls.

### CODE REVIEW

File: spec/controllers/courses_controller_spec.rb [hunk 1]
```diff
@@ -172,7 +172,7 @@
 
     context 'updating course visibility' do
       context 'as an authorized instructor' do
-        it 'responds with success on update' do
+        it 'responds with 302 on update' do
```
Comment (david-yz-liu) [inline spec/controllers/courses_controller_spec.rb:175]: this shouldn't be changed

**Review by david-yz-liu — COMMENTED** [review]
In terms of testing: the changes you made to the `course_policy_spec.rb` file are good. In addition to this, you should add some new tests to `courses_controller_spec.rb` for your updates. You can use `expect(response).to redirect_to <path>` to assert that the route redirects to the correct location.

File: app/controllers/courses_controller.rb [hunk 2]
```diff
@@ -25,15 +25,22 @@ def index
   def edit; end
 
   def update
+    if current_role.student? || current_role.ta?
```
Comment (david-yz-liu) [inline app/controllers/courses_controller.rb:28]: @WeissSa You can revert this change as well (since you've now removed the change to the corresponding test)

Comment (WeissSa) [reply app/controllers/courses_controller.rb]: Oops. Thought I did but must have forgotten to add it.

**Review by david-yz-liu — APPROVED** [review]
